### PR TITLE
[DNM] Add rawkv client

### PR DESCRIPTION
**Pull request by birdstorm**
Add a raw java client that supports:
- Put
- Get
- Scan
- Delete

Note: This PR is moved into an independent repository at https://github.com/tikv/client-java

### CODE REVIEW

**Comment by ilovesoup** [comment]
Let's start review first while perf test is ongoing.

File: tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java [hunk 1]
```diff
@@ -200,13 +138,124 @@ private ByteString getHelper(GetResponse resp) {
...
+            .setContext(context)
+            .setStartKey(startKey)
+            .setEndKey(endKey)
+            .build();
+
+    KVErrorHandler<RawDeleteRangeResponse> handler =
+        new KVErrorHandler<>(
+            regionManager, this, region, resp -> resp.hasRegionError() ? resp.getRegionError() : null);
+    RawDeleteRangeResponse resp = callWithRetry(backOffer, TikvGrpc.METHOD_RAW_DELETE_RANGE, factory, handler);
+    if (resp == null) {
```
Comment (Handora) [inline tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java]: resp is never null, the same as below

File: tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java [hunk 2]
```diff
@@ -200,13 +138,124 @@ private ByteString getHelper(GetResponse resp) {
...
+    return rawBatchScan(backOffer, key, getConf().getScanBatchSize());
+  }
+
+  public List<KvPair> rawBatchScan(BackOffer backOffer, ByteString key, int limit) {
+    return rawBatchScan(backOffer, key, limit, false);
+  }
+
+  private List<KvPair> rawBatchScanHelper(RawScanResponse resp) {
+    if (resp.hasRegionError()) {
+      throw new RegionException(resp.getRegionError());
```
[the remark this answers is not in the document]
Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java]: scan will not occur keyerror

File: tikv-client/src/main/java/com/pingcap/tikv/region/RegionManager.java [hunk 3]
```diff
@@ -227,8 +254,9 @@ public boolean updateLeader(long regionId, long storeId) {
 
   /**
    * Clears all cache when a TiKV server does not respond
+   *
    * @param regionId region's id
-   * @param storeId TiKV store's id
+   * @param storeId  TiKV store's id
```
[the remark this answers is not in the document]
Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/region/RegionManager.java]: formatted

**Comment by birdstorm** [comment]
Closed since the PR is moved into another independent repository.

**Pull request closed without merging**